src/main/java/edu/harvard/iq/dataverse/DataverseServiceBean.java

```diff
@@ -505,35 +505,91 @@ public List<Dataverse> filterByAliasQuery(String filterQuery) {
         return ret;
     }
     
-    public List<Dataverse> filterDataversesForLinking(String query, DataverseRequest req, Dataset dataset) {
+    public List<Dataverse> filterDataversesForLinking(String query, DataverseRequest req, DvObject dvo) {
```

Comment:
It looks like this method is only called from the JSF-related DatasetPage class, and furthermore, only dataset dv objects are ever sent. So, if I'm not mistaken, your changes would never be used. What's the reason for this update, if it's related to JSF and also never gets executed?

Reply:
Yes. In my first pass at this for the api, I modified this method to take in either a dataset or dataverse since that was what the api would pass - then I applied the permissions. After discussing with German, I realized that getting the collections the user had permissions on first would be more efficient. I left the method as I had modified it since it was doing what it did before for datasets - and if we wanted the dataverse version to do the same for jsf it would be ready.